File: spec/journey/find_non_domestic_certificate_by_rrn_spec.rb
# frozen_string_literal: true

describe 'Find Non-Domestic Certificate By RRN in English', journey: true do
  context 'when searching for a domestic certificate' do
    before do
      visit find_service
      click_on 'Start now'
      find('#label-non-domestic').click
      click_on 'Continue'
      click_on 'find a certificate by using its certificate number'
      fill_in 'reference_number', with: '0060-8260-7119-7386-8570'
      click_button 'Find'
    end

    it 'shows the certificate with the expected header' do
      expect(page).to have_content 'Display energy certificate (DEC)'
    end
  end
end

describe 'Find Non-Domestic Certificate By Postcode in Welsh', journey: true do
  context 'when searching for a domestic certificate in Welsh' do
    before do
      visit find_service_welsh
      click_on 'Dechrau nawr'
      find('#label-non-domestic').click
      click_on 'Parhau'
      click_on 'chwilio am dystysgrif drwy ddefnyddio’i rhif tystysgrif'
      fill_in 'reference_number', with: '0060-8260-7119-7386-8570'
      click_button 'Chwiliwch'
    end

    it 'shows the certificate with the expected header' do
      expect(page).to have_content 'Tystysgrif ynni i’w harddangos (DEC)'
    end
  end
end

File: cypress/integration/find_non_domestic_certificate_by_rrn_spec.js
describe('Find non-domestic certificate by RRN (English)', () => {
  beforeEach(() => {
    cy.visit(Cypress.env(`find_service_en_${Cypress.env('API_STAGE') || 'production'}`))
    cy.contains('Start now').click()
    cy.get('#label-non-domestic').click()
    cy.contains('button', 'Continue').click()
    cy.contains('find a certificate by using its certificate number').click()
    cy.get('input[name=reference_number]').type('0060-8260-7119-7386-8570')
    cy.contains('button', 'Find').click()
  })

  it('shows the certificate with the expected header', () => {
    cy.get('body').should('contain', 'Display energy certificate (DEC)')
  })
})

describe('Find non-domestic certificate by postcode (Welsh)', () => {
  beforeEach(() => {
    cy.visit(Cypress.env(`find_service_cy_${Cypress.env('API_STAGE') || 'production'}`))
    cy.contains('Dechreuwch nawr').click()
    cy.get('#label-non-domestic').click()
    cy.contains('Parhau').click()
    cy.contains('chwilio am dystysgrif drwy ddefnyddio’i rhif tystysgrif').click()
    cy.get('input[name=reference_number]').type('0060-8260-7119-7386-8570')
    cy.contains('button', 'Chwiliwch').click()
  })

  it('shows the certificate with the expected header', () => {
    cy.get('body').should('contain', 'Tystysgrif ynni i’w harddangos (DEC)')
  })
})
